src/arithmetic/bound_deducer.cc

```diff
@@ -297,6 +331,18 @@ void BoundDeducer::Transform() {
 void BoundDeducer::Deduce() {
   Init();
   if (!success_) return;
+
+  // Any variable appears in both expr and result,
```

Comment:
I do not understand the following comment. Why would n/4 be simplified to 0? if n>=0 the simplification rule does not stand, if n< 4, then it is a valid simplification

Reply:
for `>=`, current logic is to first get the lower bound of the left side then deduce for `i`.

Reply:
I see, perhaps we can add more explanation then. 

Also, would it results in an incorrect bound? or just a more strict bound(which still makes the condition holds)

For a better approach. My feeling is that we should first have a rewriting that tries to move vars into one side

Reply:
I believe it is just a more strict bound.
I agree, let me have a try to see if it is feasible.

Reply:
as it is also using `IntervalSetEvaluator`, this is also not required after having shape_var return single_pt